### ISSUE TITLE
Cannot activate rust-analyzer: bootstrap error.

### ISSUE THREAD

**Issue by Pelfox** [rust-lang/rust-analyzer#12876]
Hi. I'm getting an error while trying to use `rust-analyzer`.

This is popup message:
```log
Cannot activate rust-analyzer: bootstrap error. See the logs in "OUTPUT > Rust Analyzer Client" (should open automatically). To enable verbose logs use { "rust-analyzer.trace.extension": true }
```

Full logs:
```log
INFO [7/25/2022, 10:46:58 PM]: Extension version: 0.2.853
INFO [7/25/2022, 10:46:58 PM]: Using configuration {
  cargoRunner: null,
  runnableEnv: null,
  inlayHints: {
    enable: true,
    smallerHints: true,
    chainingHints: true,
    maxLength: 25,
    parameterHints: true,
    typeHints: true,
    hideNamedConstructorHints: false
  },
  updates: { channel: 'stable', askBeforeDownload: false },
  server: { path: null, extraEnv: null },
  trace: { server: 'off', extension: false },
  debug: {
    engine: 'auto',
    sourceFileMap: {
      '/rustc/<id>': '${env:USERPROFILE}/.rustup/toolchains/<toolchain-id>/lib/rustlib/src/rust'
    },
    openDebugPane: false,
    engineSettings: {}
  },
  assist: {
    importGranularity: 'crate',
    importEnforceGranularity: false,
    importPrefix: 'plain',
    importGroup: true,
    allowMergingIntoGlobImports: true
  },
  cache: { warmup: true },
  callInfo: { full: true },
  cargo: {
    autoreload: true,
    allFeatures: false,
    unsetTest: [ 'core' ],
    features: [],
    runBuildScripts: true,
    useRustcWrapperForBuildScripts: true,
    noDefaultFeatures: false,
    target: null,
    noSysroot: false
  },
  checkOnSave: {
    enable: true,
    allFeatures: null,
    allTargets: true,
    command: 'check',
    noDefaultFeatures: null,
    target: null,
    extraArgs: [],
    features: null,
    overrideCommand: null
  },
  completion: {
    addCallArgumentSnippets: true,
    addCallParenthesis: true,
    snippets: {},
    postfix: { enable: true },
    autoimport: { enable: true },
    autoself: { enable: true }
  },
  diagnostics: {
    enable: true,
    enableExperimental: true,
    disabled: [],
    remapPrefix: {},
    warningsAsHint: [],
    warningsAsInfo: []
  },
  experimental: { procAttrMacros: true },
  files: { watcher: 'client', excludeDirs: [] },
  highlightRelated: {
    references: true,
    exitPoints: true,
    breakPoints: true,
    yieldPoints: true
  },
  highlighting: { strings: true },
  hover: { documentation: true, linksInHover: true },
  hoverActions: {
    debug: true,
    enable: true,
    gotoTypeDef: true,
    implementations: true,
    references: false,
    run: true
  },
  joinLines: {
    joinElseIf: true,
    removeTrailingComma: true,
    unwrapTrivialBlock: true,
    joinAssignments: true
  },
  lens: {
    debug: true,
    enable: true,
    implementations: true,
    run: true,
    methodReferences: false,
    references: false,
    enumVariantReferences: false,
    forceCustomCommands: true
  },
  linkedProjects: [],
  lruCapacity: null,
  notifications: { cargoTomlNotFound: true },
  procMacro: { enable: true, server: null },
  runnables: { overrideCargo: null, cargoExtraArgs: [] },
  rustcSource: null,
  rustfmt: {
    extraArgs: [],
    overrideCommand: null,
    enableRangeFormatting: false
  },
  workspace: { symbol: { search: { scope: 'workspace', kind: 'only_types' } } }
}
INFO [7/25/2022, 10:46:58 PM]: PersistentState: {
  lastCheck: undefined,
  nightlyReleaseId: undefined,
  serverVersion: '0.2.853'
}
INFO [7/25/2022, 10:47:00 PM]: Using server binary at /home/pelfox/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu
WARN [7/25/2022, 10:47:00 PM]: /home/pelfox/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu --version: {
  error: [Error: spawnSync /home/pelfox/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu EACCES
  	at Object.spawnSync (node:internal/child_process:1083:20)
  	at spawnSync (node:child_process:821:24)
  	at Yy (/home/pelfox/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:61:4063)
  	at L0 (/home/pelfox/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:109:3680)
  	at processTicksAndRejections (node:internal/process/task_queues:96:5)
  	at async F0 (/home/pelfox/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:109:1881)
  	at async O0 (/home/pelfox/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:108:3977)
  	at async lC (/home/pelfox/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:108:3808)
  	at async b._activate (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:8180)
  	at async b._waitForDepsThenActivate (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:8122)
  	at async b._initialize (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:7486)] {
    errno: -13,
    code: 'EACCES',
    syscall: 'spawnSync /home/pelfox/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu',
    path: '/home/pelfox/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu',
    spawnargs: [ '--version' ]
  },
  status: null,
  signal: null,
  output: null,
  pid: 0,
  stdout: null,
  stderr: null
}
ERROR [7/25/2022, 10:47:00 PM]: Bootstrap error [Error: Failed to execute /home/pelfox/.config/Code - OSS/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu --version
	at L0 (/home/pelfox/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:109:3922)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async F0 (/home/pelfox/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:109:1881)
	at async O0 (/home/pelfox/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:108:3977)
	at async lC (/home/pelfox/.vscode-oss/extensions/matklad.rust-analyzer-0.2.853/out/main.js:108:3808)
	at async b._activate (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:8180)
	at async b._waitForDepsThenActivate (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:8122)
	at async b._initialize (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:85:7486)]
```

I'm using `Linux arch 5.18.14-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 23 Jul 2022 11:46:17 +0000 x86_64 GNU/Linux` with VS Code `1.69.2`.

**Comment by lnicola** [comment]
Duplicate of #11080, try downloading a VSIX from the GitHub Releases page.

**Comment by LintianShi** [comment]
I try the solution of issue 11080, but there is still an error.
```
INFO [7/26/2022, 8:13:43 PM]: Using server binary at /home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/server/rust-analyzer
ERROR [7/26/2022, 8:13:43 PM]: Bootstrap error [Error: Failed to execute /home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/server/rust-analyzer --version
	at OS (/home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/out/main.js:85:2289)
	at async xS (/home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/out/main.js:84:1060)
	at async dv (/home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/out/main.js:84:891)
	at async w._activate (/home/lintianshi/.vscode-server/bin/92d25e35d9bf1a6b16f7d0758f25d48ace11e5b9/out/vs/workbench/api/node/extensionHostProcess.js:85:8180)
	at async w._waitForDepsThenActivate (/home/lintianshi/.vscode-server/bin/92d25e35d9bf1a6b16f7d0758f25d48ace11e5b9/out/vs/workbench/api/node/extensionHostProcess.js:85:8122)
	at async w._initialize (/home/lintianshi/.vscode-server/bin/92d25e35d9bf1a6b16f7d0758f25d48ace11e5b9/out/vs/workbench/api/node/extensionHostProcess.js:85:7486)]
```

**Comment by lnicola** [comment]
@LintianShi What distro are you running, and do you get an error message if you try to run `/home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/server/rust-analyzer --version`?

**Comment by LintianShi** [comment]
```
/home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/server/rust-analyzer --version
/home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/server/rust-analyzer: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by /home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/server/rust-analyzer)
/home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/server/rust-analyzer: /lib64/libc.so.6: version `GLIBC_2.18' not found (required by /home/lintianshi/.vscode-server/extensions/rust-lang.rust-analyzer-0.3.1140/server/rust-analyzer)
```
I think it is due to the lack of libc6 :)

**Comment by lnicola** [comment]
You can build it from source. `cargo xtask install --server` only takes a couple of minutes and will copy the server to `~/.cargo/bin`, then you can point to it in the settings. You should probably have a matching VSIX.

If you also want to build the Code extension, remove `--server`, but you'll need Node 16 and npm.

**Comment by Vincenzocr** [comment]
I have the same bootstrap error with rust analyzer.
I am not an IT guy... might be a proxy issue? or that the vs-code version is kept fixed from the organization?

Here the full log, thank you in advance:
ERROR [3/3/2023, 18:05:22]: Bootstrap error [Error: Failed to execute c:\Users\mySelf\.vscode\extensions\rust-lang.rust-analyzer-0.4.1422-win32-x64\server\rust-analyzer.exe --version
	at Xv (c:\Users\mySelf\.vscode\extensions\rust-lang.rust-analyzer-0.4.1422-win32-x64\out\main.js:83:4925)
	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async ba.getOrCreateClient (c:\Users\mySelf\.vscode\extensions\rust-lang.rust-analyzer-0.4.1422-win32-x64\out\main.js:96:2023)
	at async ba.start (c:\Users\mySelf\.vscode\extensions\rust-lang.rust-analyzer-0.4.1422-win32-x64\out\main.js:96:2936)
	at async X_ (c:\Users\mySelf\.vscode\extensions\rust-lang.rust-analyzer-0.4.1422-win32-x64\out\main.js:100:1948)
	at async G_ (c:\Users\mySelf\.vscode\extensions\rust-lang.rust-analyzer-0.4.1422-win32-x64\out\main.js:100:763)
	at async E._activate (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:85:8224)
	at async E._waitForDepsThenActivate (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:85:8166)
	at async E._initialize (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:85:7530)]
INFO [3/3/2023, 18:07:58]: Extension version: 0.4.1422
INFO [3/3/2023, 18:07:58]: Using configuration {
  cargoRunner: null,
  runnableEnv: null,
  server: { path: null, extraEnv: null },
  trace: { server: 'off', extension: true },
  debug: {
    engine: 'auto',
    sourceFileMap: {
      '/rustc/<id>': '${env:USERPROFILE}/.rustup/toolchains/<toolchain-id>/lib/rustlib/src/rust'
    },
    openDebugPane: false,
    engineSettings: {}
  },
  restartServerOnConfigChange: false,
  typing: {
    continueCommentsOnNewline: true,
    autoClosingAngleBrackets: { enable: false }
  },
  diagnostics: {
    previewRustcOutput: false,
    useRustcErrorCode: false,
    disabled: [],
    enable: true,
    experimental: { enable: false },
    remapPrefix: {},
    warningsAsHint: [],
    warningsAsInfo: []
  },
  assist: { emitMustUse: false, expressionFillDefault: 'todo' },
  cachePriming: { enable: true, numThreads: 0 },
  cargo: {
    autoreload: true,
    buildScripts: {
      enable: true,
      invocationLocation: 'workspace',
      invocationStrategy: 'per_workspace',
      overrideCommand: null,
      useRustcWrapper: true
    },
    extraEnv: {},
    features: [],
    noDefaultFeatures: false,
    sysroot: 'discover',
    sysrootSrc: null,
    target: null,
    unsetTest: [ 'core' ]
  },
  checkOnSave: true,
  check: {
    allTargets: true,
    command: 'check',
    extraArgs: [],
    extraEnv: {},
    features: null,
    invocationLocation: 'workspace',
    invocationStrategy: 'per_workspace',
    noDefaultFeatures: null,
    overrideCommand: null,
    targets: null
  },
  completion: {
    autoimport: { enable: true },
    autoself: { enable: true },
    callable: { snippets: 'fill_arguments' },
    limit: null,
    postfix: { enable: true },
    privateEditable: { enable: false },
    snippets: {
      custom: {
        'Arc::new': {
          postfix: 'arc',
          body: 'Arc::new(${receiver})',
          requires: 'std::sync::Arc',
          description: 'Put the expression into an `Arc`',
          scope: 'expr'
        },
        'Rc::new': {
          postfix: 'rc',
          body: 'Rc::new(${receiver})',
          requires: 'std::rc::Rc',
          description: 'Put the expression into an `Rc`',
          scope: 'expr'
        },
        'Box::pin': {
          postfix: 'pinbox',
          body: 'Box::pin(${receiver})',
          requires: 'std::boxed::Box',
          description: 'Put the expression into a pinned `Box`',
          scope: 'expr'
        },
        Ok: {
          postfix: 'ok',
          body: 'Ok(${receiver})',
          description: 'Wrap the expression in a `Result::Ok`',
          scope: 'expr'
        },
        Err: {
          postfix: 'err',
          body: 'Err(${receiver})',
          description: 'Wrap the expression in a `Result::Err`',
          scope: 'expr'
        },
        Some: {
          postfix: 'some',
          body: 'Some(${receiver})',
          description: 'Wrap the expression in an `Option::Some`',
          scope: 'expr'
        }
      }
    }
  },
  files: { excludeDirs: [], watcher: 'client' },
  highlightRelated: {
    breakPoints: { enable: true },
    exitPoints: { enable: true },
    references: { enable: true },
    yieldPoints: { enable: true }
  },
  hover: {
    actions: {
      debug: { enable: true },
      enable: true,
      gotoTypeDef: { enable: true },
      implementations: { enable: true },
      references: { enable: false },
      run: { enable: true }
    },
    documentation: { enable: true, keywords: { enable: true } },
    links: { enable: true }
  },
  imports: {
    granularity: { enforce: false, group: 'crate' },
    group: { enable: true },
    merge: { glob: true },
    prefer: { no: { std: false } },
    prefix: 'plain'
  },
  inlayHints: {
    bindingModeHints: { enable: false },
    chainingHints: { enable: true },
    closingBraceHints: { enable: true, minLines: 25 },
    closureReturnTypeHints: { enable: 'never' },
    discriminantHints: { enable: 'never' },
    expressionAdjustmentHints: { enable: 'never', hideOutsideUnsafe: false, mode: 'prefix' },
    lifetimeElisionHints: { enable: 'never', useParameterNames: false },
    maxLength: 25,
    parameterHints: { enable: true },
    reborrowHints: { enable: 'never' },
    renderColons: true,
    typeHints: {
      enable: true,
      hideClosureInitialization: false,
      hideNamedConstructor: false
    }
  },
  interpret: { tests: false },
  joinLines: {
    joinAssignments: true,
    joinElseIf: true,
    removeTrailingComma: true,
    unwrapTrivialBlock: true
  },
  lens: {
    debug: { enable: true },
    enable: true,
    forceCustomCommands: true,
    implementations: { enable: true },
    location: 'above_name',
    references: {
      adt: { enable: false },
      enumVariant: { enable: false },
      method: { enable: false },
      trait: { enable: false }
    },
    run: { enable: true }
  },
  linkedProjects: [],
  lru: { capacity: null },
  notifications: { cargoTomlNotFound: true },
  numThreads: null,
  procMacro: {
    attributes: { enable: true },
    enable: true,
    ignored: {},
    server: null
  },
  references: { excludeImports: false },
  runnables: { command: null, extraArgs: [] },
  rustc: { source: null },
  rustfmt: {
    extraArgs: [],
    overrideCommand: null,
    rangeFormatting: { enable: false }
  },
  semanticHighlighting: {
    doc: { comment: { inject: { enable: true } } },
    operator: { enable: true, specialization: { enable: false } },
    punctuation: {
      enable: false,
      separate: { macro: { bang: false } },
      specialization: { enable: false }
    },
    strings: { enable: true }
  },
  signatureInfo: { detail: 'full', documentation: { enable: true } },
  workspace: {
    symbol: { search: { kind: 'only_types', limit: 128, scope: 'workspace' } }
  }
}

**Comment by lnicola** [comment]
@Vincenzocr

> `c:\Users\mySelf.vscode\extensions\rust-lang.rust-analyzer-0.4.1422-win32-x64\server\rust-analyzer.exe --version`

Do you have that program? What do you get if you run it? There's a missing `\` there, maybe you removed it by mistake.

**Comment by lnicola** [comment]
Don't double-click, open a Comand Prompt in that directory and run `rust-analyzer --version`.

**Comment by Vincenzocr** [comment]
@lnicola it says: "rust_analyzer.exe" not recognized a as  an internal or external command, an executable program or batch file
